math/src/polynomial.rs

```diff
@@ -11,12 +11,13 @@ pub struct Polynomial<FE> {
 
 impl<F: IsField> Polynomial<FieldElement<F>> {
     /// Creates a new polynomial with the given coefficients
-    pub fn new(coefficients: Vec<FieldElement<F>>) -> Self {
+    pub fn new(coefficients: &[FieldElement<F>]) -> Self {
         // Removes trailing zero coefficients at the end
         let mut unpadded_coefficients = coefficients
-            .into_iter()
+            .iter()
             .rev()
-            .skip_while(|x| *x == FieldElement::zero())
+            .skip_while(|x| **x == FieldElement::zero())
```

Comment:
```suggestion
            .skip_while(|x| x.is_zero())
```

Reply:
The original code will create a new `FieldElement` for every iteration, while the `is_zero()` method does the check without creating new objects that could potentially allocate.